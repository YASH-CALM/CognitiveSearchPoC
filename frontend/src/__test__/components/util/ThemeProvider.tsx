import * as React from 'react'
import { Theme as UwpThemeProvider } from 'react-uwp/Theme';
import { theme } from '../../../components/App/theme';
export const withTheme = (children: string | JSX.Element) => <UwpThemeProvider theme={theme}>{children}</UwpThemeProvider>