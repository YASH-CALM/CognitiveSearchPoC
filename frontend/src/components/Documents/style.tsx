/**
 * 
 * @param number total number of pages in the document
 */
export const documentsStyle = (totalPages: number): React.CSSProperties => ({
    height: totalPages > 1 ? 'calc(100% - 50px)' : '100%',
    overflowY: 'auto',
    position: 'relative',
    top: totalPages > 1 ? 50 : 0,
})
export const pagenationStyle: React.CSSProperties = {
    borderTop: '2px solid #88888888',
    display: 'flex',
    height: 40,
    justifyContent: 'center',
    margin: '0 auto',
    width: '100%'
}

/**
 * 
 * @param theme the react-uwp theme
 * @param width width of the pagenation bar
 */
export const pagenationBarStyle = (theme: any, width?: string | number): React.CSSProperties => ({
    ...theme.typographyStyles.Header,
    backgroundColor: '#222',
    height: 50,
    position: 'fixed',
    top: 48,
    transition: 'width .25s ease-in-out',
    width,
    zIndex: 90001,
})