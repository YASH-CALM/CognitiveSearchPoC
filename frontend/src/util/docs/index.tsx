import { IDocument } from 'src/types/documents';

export const rNumberRange = (max: number, min?: number): number => {
    return Math.floor(Math.random() * max) + (min === 0 ? 0 : (min || 1));
}

export const rLetter = (): string => {
    return 'abcdefghijklmnopqrstuvwxyz'[rNumberRange(26, 0)];
}

const loremIpsum: string[] = `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
Integer sed molestie quam, eleifend imperdiet turpis.
In ut mauris enim. Curabitur eget tempus leo. Etiam
faucibus placerat vulputate. Suspendisse potenti.
In risus velit, scelerisque maximus vulputate vitae,
mollis sed mi. In eleifend, tortor in ultricies pulvinar,
lorem ipsum tincidunt eros, ut dignissim orci ipsum ut
tortor. Mauris porta lorem non nunc malesuada, sed euismod
metus gravida. Morbi tempus ante ac cursus finibus.
Phasellus elementum ac justo ut rutrum. Aliquam id tempus
tellus, vitae consectetur leo. Mauris sapien urna, feugiat
sed nulla id, dictum pharetra velit. Quisque accumsan
blandit orci nec sollicitudin. Phasellus vel ipsum consequat
arcu finibus pretium nec ut leo.`.split(' ');

export const rTag = (): string => loremIpsum[rNumberRange(loremIpsum.length-1, 0)];

export const rDocument = (): IDocument => ({
    pages: Array(rNumberRange(5)).fill(0).map(m => rNumberRange(5)).map(p => ({
        source: `/docs/phb/${p}.png`,
        tags: Array(rNumberRange(26)).fill(0).map(rTag),
    })),
    title: `${rLetter()}${rLetter()}${rLetter()}_${rNumberRange(1000000000)}`,
})

export const docs = (): IDocument[] => Array(rNumberRange(50,25)).fill(0).map(rDocument);