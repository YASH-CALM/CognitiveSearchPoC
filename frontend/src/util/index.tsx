export const isArrayEmpty = (array?: any[]): boolean => !array || !array.length;
export const isDefinedNumber = (n?: number) : boolean => n != null;