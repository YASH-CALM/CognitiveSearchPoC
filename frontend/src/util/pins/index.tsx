import { IInfoboxWithPushPin } from 'src/types/pins';

export const pins: IInfoboxWithPushPin[] = [
    {
        infobox: {
            description: 'this location was entered as a longitude latitude pair "[36, -112]"',
            title: 'Grand Canyon',
        },
        location: [36, -112],
        pushPin: {
            description: 'Grand Canyon',
            options: { color: 'orange' },
            title: 'Grand Canyon',
        }
    },
    {
        infobox: {
            description: 'this location was entered as an address "841 Avenue of the Cities"',
            title: 'YASH Office',
        },
        location: ['841 Avenue of the Cities'],
        pushPin: {
            description: 'Yash Office',
            options: { color: '#005ba1' },
            title: 'YASH',
        }
    },
    {
        infobox: {
            description: 'this location was entered as a longitude latitude pair "[45, -84]"',
            title: 'Lake Superior',
        },
        location: [45, -84],
        pushPin: {
            description: 'Lake Superior',
            options: { color: 'lightgreen' },
            title: 'Lake Superior',
        }
    },
    {
        infobox: {
            description: 'this location was entered as the name of the location "Statue of Liberty"',
            title: 'Statue of Liberty',
        },
        location: ['Statue of Liberty'],
        pushPin: {
            description: 'Statue of Liberty',
            options: { color: 'grey' },
            title: 'Statue of Liberty',
        }
    },
]