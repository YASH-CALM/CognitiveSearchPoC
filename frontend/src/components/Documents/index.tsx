import * as PropTypes from 'prop-types';
import * as React from 'react';
import { IconButton } from 'react-uwp';
import { IDocument } from '../../types/documents';
import DocumentPreview from '../DocumentPreview';
import * as Style from './style';

export interface IDocumentsProps extends React.HTMLAttributes<HTMLDivElement> {
    docs: IDocument[],
    pageSize: number,
    pagenationWidth: string | number,
    pagePadding: number,
}

export interface IDocumentsState {
    currentPage: number,
}

const pagenator = (currentPage: number, padding: number, totalPages: number) => {
    if (2 * padding + 1 >= totalPages) {
        return Array(totalPages).fill(0).map((p, i) => i + 1);
    } else {
        const min = Math.max(1, currentPage + padding + 1 < totalPages ? currentPage - padding + 1 : totalPages - (2 * padding));
        return Array(2 * padding + 1).fill(min).map((p, i) => p + i);
    }
}

export default class Documents extends React.Component<IDocumentsProps, IDocumentsState> {
    public static contextTypes = { theme: PropTypes.object };
    constructor(props: IDocumentsProps) {
        super(props);
        this.state = {
            currentPage: 0,
        }
    }

    public componentWillReceiveProps(nextProps: IDocumentsProps) {
        if (this.props.docs.length > nextProps.docs.length) {
            this.setState({ currentPage: 0 });
        }
    }

    public setCurrentPage = (currentPage: number) => () => {
        this.setState({ currentPage })
    }
    public render() {
        const { theme } = this.context;
        const { docs, pagenationWidth, pagePadding, pageSize } = this.props;
        const totalPages: number = Math.ceil(docs.length / pageSize);
        const documentsStyle: React.CSSProperties = Style.documentsStyle(totalPages);
        const pagenationStyle: React.CSSProperties = Style.pagenationStyle;
        const pagenationBarStyle: React.CSSProperties = Style.pagenationBarStyle(theme, pagenationWidth);
        const pageIndices: number[] = pagenator(this.state.currentPage, pagePadding, totalPages);
        return (
            <>
                {
                    totalPages > 1 ?
                        <div style={pagenationBarStyle}>
                            <div style={pagenationStyle}>
                                <IconButton disabled={this.state.currentPage <= 0} onClick={this.setCurrentPage(this.state.currentPage > 0 ? this.state.currentPage - 1 : this.state.currentPage)}>ChevronLeft</IconButton>
                                {pageIndices.map(index => <IconButton key={index} onClick={this.setCurrentPage(index - 1)} disabled={(index - 1) === this.state.currentPage}>{index}</IconButton>)}
                                <IconButton disabled={(this.state.currentPage + 1) >= totalPages} onClick={this.setCurrentPage((this.state.currentPage + 1) < totalPages ? this.state.currentPage + 1 : this.state.currentPage)}>ChevronRight</IconButton>
                            </div>
                        </div>
                        : undefined
                }
                <div style={documentsStyle}>
                    {
                        docs.slice(this.state.currentPage * pageSize, (this.state.currentPage + 1) * pageSize).map(doc => <DocumentPreview key={doc.title} style={this.props.style} doc={doc} />)
                    }
                </div>
            </>
        );
    }
}
