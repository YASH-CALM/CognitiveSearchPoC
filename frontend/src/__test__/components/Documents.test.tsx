import * as React from 'react'
import { createRenderer as ShallowRenderer } from 'react-test-renderer/shallow'
import Documents from '../../components/Documents'
import { withTheme } from './util/ThemeProvider'

describe('Documents Component', () => {
    const renderer = ShallowRenderer()
    const props = {
        docs: 'abcd'.split('').map(d => ({title: d, pages: []})),
        pagePadding: 2,
        pageSize: 10,
        pagenationWidth: '100%',
    }
    it('should match snapshot', () => {
        renderer.render(withTheme(<Documents {...props}/>))
        const result = renderer.getRenderOutput()
        expect(result).toMatchSnapshot()
    })
})