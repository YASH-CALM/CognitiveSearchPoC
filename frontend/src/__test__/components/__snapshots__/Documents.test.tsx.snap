// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documents Component should match snapshot 1`] = `
<div
  className="theme-root-2366926892 react-uwp-theme-dark"
>
  <RenderToBody
    className="fluent-background-3499734374"
  />
  <RenderToBody>
    <ToastWrapper />
  </RenderToBody>
  <Documents
    docs={
      Array [
        Object {
          "pages": Array [],
          "title": "a",
        },
        Object {
          "pages": Array [],
          "title": "b",
        },
        Object {
          "pages": Array [],
          "title": "c",
        },
        Object {
          "pages": Array [],
          "title": "d",
        },
      ]
    }
    pagePadding={2}
    pageSize={10}
    pagenationWidth="100%"
  />
</div>
`;
